### ISSUE TITLE
Can't get the reported result in paper

### ISSUE THREAD

**Issue by greatlog** [open-mmlab/mmsr#20]
Sorry, I've got a problem. I  train RCAN , RDN and EDSR according to your setting. Data preprocess is also done by your scripts. But I can't get the reported results in those papers. Empirically there is always 0.09dB difference. I wonder if you have this difference too.  If so, Is that a problem about data preprocess?  Thank you.

**Comment by xinntao** [comment]
I have not tested these algorithms in mmsr. 
For EDSR, do you also encounter 0.09dB performance drop?
In mmsr, we use a variant of EDSR_baseline_x4 (1.52 M) (namely, the modified SRResNet) and we can get the comparable (slightly better) performance. 

If you train a large model, the performance drop may arise from the data or the optimization strategy.

**Comment by m732367606** [comment]
Sorry, I've got a problem about the training iteration. I am trying to reproduce the results of ESRGAN_PSNR_oriented model but I have no idea how many iterations I should set to get optimal results.
To be honest, I have read the paper of ESRGAN muptiple times, but it seems that there is no exact iteration numbers in it, only said, 'The learning rate is initialized as 2e-4 and decayed by a factor of 2 every 2e5 of mini-batch updates'.
Hope Professor Xintao or greatlog can share the iterations's setting of ESRGAN_PSNR_oriented model when the batch_size is 16 to save my aimless trying times. Thank you!

**Comment by John-Yao** [comment]
@greatlog 
Hello, do you modify EDSR from EDSR-Pytorch. I noticed that in EDSR-Pytorch repo, the data fed into model has range of [0,255] but in the repo the data has range of [0,1]. Did you compare the result of such different setting in mmsr? Thanks!

**Comment by ascetic-yuanma** [comment]
> @greatlog
> Hello, do you modify EDSR from EDSR-Pytorch. I noticed that in EDSR-Pytorch repo, the data fed into model has range of [0,255] but in the repo the data has range of [0,1]. Did you compare the result of such different setting in mmsr? Thanks!

I tested last week, there is no big difference between range [0,1] and range [0,255], the PSNR and SSIM results are about the same.

**Comment by John-Yao** [comment]
@ascetic-yuanma Thanks for you reply!

**Comment by John-Yao** [comment]
@ascetic-yuanma Could you successfully reproduce EDSR model in mmsr, I copy EDSR from EDSR-pytorch and modify rgb_range from 255 to 1.0. but also  encounter about 0.09dB performance drop even with same  configs and dataset.

**Comment by ascetic-yuanma** [comment]
@John-Yao Hi John, I did some extending work these days, and found there's a huge drop of PSNR on Set5 at scaling ratio 4 when reproducing MSRN on this framework.
Futhermore, I noticed a PSNR&SSIM drop when calculating PSNR on Matlab using default psnr and ssim functions.
For instance, the PSNR&SSIM of my reproduced MSRN calculated by mmsr are 32.10dB/0.8934, while the PSNR&SSIM of my reproduced MSRN calculated by matlab are 32.07dB/0.8923.

**Comment by ascetic-yuanma** [comment]
@John-Yao The reported PSNR&SSIM of MSRN on Set5_X4 are 32.25/0.8958 collected from https://github.com/MIVRC/MSRN-PyTorch.

**Comment by ascetic-yuanma** [comment]
@John-Yao Will keep you informed if figured out what caused this. Hope we can figure it out!

**Comment by ascetic-yuanma** [comment]
Forgot to @greatlog  :) Hope we can figure this out!

**Comment by John-Yao** [comment]
I think the drop of PSNR/SSIM is due to the code of PSNR function by different repo. 
In my exp,   PSNR of my reproduced EDSR is slightly higher than origin repo using the same evaluation function.
About MSRN, I think is due to some different settings such as lr scheduler...

**Comment by ascetic-yuanma** [comment]
@John-Yao So did you successfully reproduce EDSR model in mmsr and get reported PSNR in EDSR paper?

**Comment by John-Yao** [comment]
@ascetic-yuanma I didn't  run exp on public dataset. In our dataset, my reproduced EDSR in mmsr repo is slightly higher than the origin repo(using the same evaluation function).

**Comment by ascetic-yuanma** [comment]
@John-Yao Sure, thank you for your reply!
